### PR TITLE
Fix: Php errors from enabled comments inside the Donor Wall have been corrected

### LINKED ISSUE #6477: Using the Donor Wall shortcode or block, while comments are enabled should not throw a PHP warning.
## Details

In GiveWP version 2.21.0 give_donor_wall shows a PHP warning if comments are not disabled.
With comments enabled, the PHP warning is shown and no comment is shown.
Some sites are not showing the warning.
Disabling debug in wp-config does not hide the warning.

## Expected Behavior

- There should be no errors when enabling show comments.
- Comments should display when enabled.

## Steps to Reproduce
- version 2.21.0 of GiveWP
- user the donor wall block or shortcode
- enable Show Comments from the inspector controls

## Visuals

https://user-images.githubusercontent.com/75056371/174894208-9d9ec094-68bf-40b6-b683-8dfb8073ac30.mov

## Acceptance Criteria

-   [x] Enabling show comments should display donation comments.
-   [x] Enabling show comments should not display errors.
-   [x] Disabling show comments should not display donation comments.

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
Resolves #6477

## Description

This PR includes several fixes for the Donor Wall. It fixes two Php warnings/errors from within the Donor Wall, includes some style updates for the block and improves the functionality that displays comments. 

We found that the test donation gateway was not saving  anonymous donations to the database.  This caused an error for donations marked as anonymous on the front end. 

We also found that if the comment length set by the inspector control was higher than the actual length of the comment, it would not display.

- Undefined variable $donor has been corrected by being properly passed to the template.
- Undefined index ['_give_anonymous_donation'] has been corrected by being properly saved to the database.
- Donation amounts will no longer wrap to the next line.
- Comments now display regardless of the character length. 

## Affects

- Donor Wall
- Test gateway

## Visuals

https://user-images.githubusercontent.com/75056371/175374410-b6e16924-4a16-4667-bae4-0a980eb8822a.mov

https://user-images.githubusercontent.com/75056371/175408178-76277c9a-6b84-477d-b318-7c700f0a9304.mov

## Testing Instructions

- Create a Donor Wall block
- Toggle 'Show Comments' on & view page for thrown errors.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Review by jonwaldstein — APPROVED** [review]
@JoshuaHungDinh looks good! please pass this along to QA.

**Review by Benunc — APPROVED** [review]
This has been tested, and confirmed that all fixes work, as well as run through automated QA to ensure that all donation functionality still works.

File: templates/shortcode-donor-wall.php [hunk 1]
```diff
@@ -117,10 +119,16 @@
...
                             echo "<p class='give-donor-content__excerpt'>$excerpt &hellip;
-                                    <span> <a class='give-donor-content__read-more' style='color: {$primary_color}'> $read_more_text </a></span>
+                                    <span> <a class='give-donor-content__read-more' style='color: $primary_color'> $read_more_text </a></span>
                                    </p>";
 
                             echo "<p class='give-donor-content__comment'> $comment </p>";
+
+                        }
+                        else     {
+                            echo "<p class='give-donor-content__excerpt'>
```
Comment (ravinderk) [inline templates/shortcode-donor-wall.php:129]: @JoshuaHungDinh for full comment, you can use `give-donor-content__comment` class.

File: templates/shortcode-donor-wall.php [hunk 2]
```diff
@@ -87,19 +89,19 @@
...
                 && absint($atts['comment_length'])
                 && !empty($donation['donor_comment'])
-                && !$donation['_give_anonymous_donation']
             ) :
                 ?>
                 <div class="give-donor-wrapper">
                     <div class="give-donor-content"
                          style="border-color: <?php echo !empty($atts['color']) ? $atts['color'] : '#219653' ?>">
                         <?php
                         $comment = trim($donation['donor_comment']);
```
Comment (ravinderk) [inline templates/shortcode-donor-wall.php:98]: @JoshuaHungDinh can we escape comment value before display on forntend?

File: includes/gateways/manual.php [hunk 3]
```diff
@@ -97,6 +97,8 @@ function give_manual_payment( $purchase_data ) {
 	$payment = give_insert_payment( $payment_data );
 
 	if ( $payment ) {
+        $is_anonymous = isset($purchase_data['post_data']['give_anonymous_donation']) && absint($purchase_data['post_data']['give_anonymous_donation']);
+        update_post_meta($payment, '_give_anonymous_donation', $is_anonymous);
```
Comment (ravinderk) [inline includes/gateways/manual.php:101]: @JoshuaHungDinh `_give_anonymous_donation ` metadata value should be an integer when stored in a database.

ref: https://github.com/impress-org/givewp/blob/b68aab707f5c42e339a8d8327b2b6bf2ba72ba17/src/Donations/Repositories/DonationRepository.php#L348

**Review by ravinderk — CHANGES_REQUESTED** [review]
@JoshuaHungDinh I requested few changes. Let me know if you have any questions.

**Review by ravinderk — APPROVED** [review]

**Pull request merged**